popup.js

```diff
@@ -67,18 +69,94 @@ function outputHTMLString(messages) {
  *     scripts.
  * @param {number} i The index of messages to use.
  */
-function fillRemainingHoles(messages, i) {
+function fillRemainingHolesAndMinimizeStyles(messages, i) {
   var html = messages[i].html;
   var frameHoles = messages[i].frameHoles;
   for (var index in frameHoles) {
     if (frameHoles.hasOwnProperty(index)) {
       var frameIndex = frameHoles[index];
       for (var j = 0; j < messages.length; j++) {
         if (messages[j].frameIndex == frameIndex) {
-          fillRemainingHoles(messages, j);
+          fillRemainingHolesAndMinimizeStyles(messages, j);
```

Comment:
I think it's okay to skip the pseudo element stuff for now.

Could you process the html element though? Without that, a simple hello world webpage still has a huge block of redundant style.